api/envoy/api/v2/core/health_check.proto

```diff
@@ -63,6 +63,15 @@ message HealthCheck {
 
   // [#comment:next free field: 10]
   message HttpHealthCheck {
+    enum CodecClientType {
+      HTTP1 = 0;
+      HTTP2 = 1;
+      // QUIC implementation is not production ready yet. Use this enum with
+      // caution to prevent accidental execution of QUIC code. I.e. `!= HTTP2` is
+      // no longer sufficient to distinguish HTTP1 and HTTP2 traffic.
```

Comment:
is health check implemented for http3? if no add `not-implented-hide`?

Reply:
If that works for enums that'd be great!

Reply:
Also I didn't see @htuch's comment on this - do we want to move towards having a single CodecClienttype which most classes use, or do we want to redefine the enum in different places to potentially limit which values are used where?  I'd be inclined towards defining a CodecType somewhere in API where it can be reused, and moving towards using it everywhere as we add and replace codec enums but it's definitely a question for @envoyproxy/api-shepherds

Reply:
+1 let's define a common type. IMO it's fine to have runtime checks in various places that don't accept all the values (e.g., we can runtime fail an HC config that asks for HTTP3 until it is supported).

Reply:
I would probably define it under https://github.com/envoyproxy/envoy/tree/master/api/envoy/type in a new http.proto, or optimally somewhere in the new UDPA repo, but I will defer to @htuch on that one. 

Per @alyssawilk I would probably just use it in new places for now and add a comment in the other places to move to it in the next API version potentially.

Reply:
I think `envoy.type` is the right place for now. I think we'd want to shop this one to @envoyproxy/udpa-wg if we wanted to put it in `udpa.type`, since it's not something obvious like `TypedStruct`, there should be some review/comment period for that.

Reply:
> +1 let's define a common type. 

Moved this enum to api/envoy/type/http.proto.

> IMO it's fine to have runtime checks in various places that don't accept all the values (e.g., we can runtime fail an HC config that asks for HTTP3 until it is supported).

If I run time fail HTTP3 value in http_connection_manager/config.cc, how can I plug in the actual quic codec in this PR?

Reply:
I was talking about runtime failing just for HTTP health check, basically places where we don't support it yet...